File: itemClass.cpp
// ============================================================================
//                               itemClass.cpp
// ============================================================================
// Programmed by: Diego Borjas
// Date Created: 2/26/2025
// CSUF Software Engineering Class
// Professor Paul Oginni

#include "itemClass.h"

// =========================== Default Constructor =========================
Item::Item()
{
	itemName = "";
	description = "";
	seller = "";
	itemPrice = 0;
	shippingCost = 0;
	itemID = 0;
	stockQuantity = 0;
	inStock = false;
}
// ============================================================================

// ========================== Overloaded Constructor ==========================
Item::Item(string itemName, string description, string seller, double itemPrice, double shippingCost, int itemID, int stockQuantity, bool inStock)
{
	this->itemName = itemName;
	this->description = description;
	this->seller = seller;
	this->itemPrice = itemPrice;
	this->shippingCost = shippingCost;
	this->itemID = itemID;
	this->stockQuantity = stockQuantity;
	this->inStock = inStock;

}
// ============================================================================


// ============================================================================

// ================================ Destructor ================================
Item::~Item(){}
// ============================================================================

// =============================== setName method =============================
// INPUTS: The item name as a string for the parameter.
// ------
// OUTPUT: No output is returned nor produced.
// -------
// DESCRIPTION: This method will set the item object class' name into its
// respective variable location.
// ============================================================================
void Item::setName(string itemName)
{
	this->itemName = itemName;
}
// ============================================================================

// ========================== setDescription method ===========================
// INPUTS: The item description as a string for the parameter.
// ------
// OUTPUT: No output is returned nor produced.
// -------
// DESCRIPTION: This method will set the item's description into its respective
// variable location.
// ============================================================================
void Item::setDescription(string description)
{
	this->description = description;
}
// ============================================================================

// ============================== setSeller method ============================
// INPUTS: The seller who sells the item as a string.
// ------
// OUTPUT: No output is returned nor produced.
// -------
// DESCRIPTION: This method will set the seller's name or company which sells
// the respective item.
// ============================================================================
void Item::setSeller(string seller)
{
	this->seller = seller;
}
// ============================================================================

// =============================== setPrice method ============================
// INPUTS: The price number as an integer.
// ------
// OUTPUT: None.
// -------
// DESCRIPTION: This method will set the item's price and store it into its
// respective variable location.
// ============================================================================
void Item::setPrice(double itemPrice)
{
	this->itemPrice = itemPrice;
}
// ============================================================================

// ============================ setShipping method ============================
// INPUTS: The shipping costs amount as a double.
// ------
// OUTPUT: None.
// -------
// DESCRIPTION: This method will set the item's shipping costs and store it
// into its respective variable location, accounted for decimal prices.
// ============================================================================
void Item::setShipping(double shippingCost)
{
	this->shippingCost = shippingCost;
}
// ============================================================================

// ============================= setID method =================================
// INPUTS: The item's ID number as an integer.
// ------
// OUTPUT: None.
// -------
// DESCRIPTION: This method will set the item's ID number and store it
// into its respective variable location, for managing and tracking purposes.
// ============================================================================
void Item::setID(int itemID)
{
	this->itemID = itemID;
}
// ============================================================================

// ========================= setQuantity method ===============================
// INPUTS: The item's quantity in-stock as an integer.
// ------
// OUTPUT: None.
// -------
// DESCRIPTION: This method will set the item's quantity amount that is
// available on the store currently and and store it into its respective
// variable location.
// ============================================================================
void Item::setQuantity(int stockQuantity)
{
	this->stockQuantity = stockQuantity;
}
// ============================================================================

// ========================= setStock method ==================================
// INPUTS: The item's status whether it is in-stock or not as a boolean.
// ------
// OUTPUT: None.
// -------
// DESCRIPTION: This method will set the item's in-stock for managing purposes
// such as to check whether the item is available for purchase or not.
// ============================================================================
void Item::setStock(bool inStock)
{
	this->inStock = inStock;
}
// ============================================================================

// ========================= printName method ==================================
// INPUTS: None.
// ------
// OUTPUT: The item object class' name will be printed.
// -------
// DESCRIPTION: This method will print the item's name.
// ============================================================================
void Item::printName() const
{
	cout << itemName << endl;

	return;
}
// ============================================================================

// ======================== printDescription method ===========================
// INPUTS: None.
// ------
// OUTPUT: The item's description will be printed.
// -------
// DESCRIPTION: This method will print the item's description.
// ============================================================================
void Item::printDescription() const
{
	cout << description << endl;

	return;
}
// ============================================================================

// ========================== printSeller method ==============================
// INPUTS: None.
// ------
// OUTPUT: The item's seller will be printed.
// -------
// DESCRIPTION: This method will print the item's seller.
// ============================================================================
void Item::printSeller() const
{
	cout << seller << endl;

	return;
}
// ============================================================================

// ============================ getPrice method ===============================
// INPUTS: None.
// ------
// OUTPUT: The item's price will be returned back to main.
// -------
// DESCRIPTION: This method will return the item's price.
// ============================================================================
double Item::getPrice() const
{
	return itemPrice;
}
// ============================================================================

// ========================== getShipping method ==============================
// INPUTS: None.
// ------
// OUTPUT: The item's shipping costs will be returned back to main.
// -------
// DESCRIPTION: This method will return the item's shipping costs.
// ============================================================================
double Item::getShipping() const
{
	return shippingCost;
}
// ============================================================================

// ============================== getID method ================================
// INPUTS: None.
// ------
// OUTPUT: The item's ID number will be returned back to main.
// -------
// DESCRIPTION: This method will return the item's ID number.
// ============================================================================
int Item::getID() const
{
	return itemID;
}
// ============================================================================

// ========================== getQuantity method ==============================
// INPUTS: None.
// ------
// OUTPUT: The item's stock quantity will be returned back to main.
// -------
// DESCRIPTION: This method will return the item's stock quantity amount.
// ============================================================================
int Item::getQuantity() const
{
	return stockQuantity;
}
// ============================================================================

// =========================== checkStock method ==============================
// INPUTS: None.
// ------
// OUTPUT: The item's stock status will be returned back to main.
// -------
// DESCRIPTION: This method will return the item's stock status.
// ============================================================================
bool Item::checkStock() const
{
	return inStock;
}
// ============================================================================

File: item_class.py
"""
Programmer: Diego Borjas
Professor: Paul Oginni
Class: Software Engineering CPSC 362
Date: 20 March 2025
Group: #4
==================================================================
Description:
The Item class will represent and create every product that will be
available for purchase in the website's catalog. Each item will have its
name, description of item, the company that sells it, its price, unique ID
for managing purposes, stock quantity available in website to sell, and a
stock status variable to check whether it is out-of-stock or not.
===================================================================
"""

class Item:
    def __init__(self, itemName="", description="", seller="", itemPrice=0.0, shippingCost=0.0, itemID=0, stockQuantity=0, inStock=False):
        self.itemName = itemName
        self.description = description
        self.seller = seller
        self.itemPrice = itemPrice
        self.shippingCost = shippingCost
        self.itemID = itemID
        self.stockQuantity = stockQuantity
        self.inStock = inStock

    # sets the item's name.
    def setName(self, itemName):
        self.itemName = itemName
        
    # sets the item's description.
    def setDescription(self, description):
        self.description = description

    # sets the name of the person or company that sells the item.
    def setSeller(self, seller):
        self.seller = seller

    # sets the item's total price.
    def setPrice(self, itemPrice):
        self.itemPrice = itemPrice

    # sets the total cost of shipping for the item.
    def setShipping(self, shippingCost):
        self.shippingCost = shippingCost

    # sets the item's ID for managing purposes. 
    def setID(self, itemID):
        self.itemID = itemID

    # sets the quantity available in-stock of the item.
    def setQuantity(self, stockQuantity):
        self.stockQuantity = stockQuantity

    # sets whether the item is in-stock or not as a boolean.
    def setStock(self, inStock):
        self.inStock = inStock

    # Retrieves the item's name. 
    def getName(self):
        return self.itemName

    # Retrieves the item's description.
    def getDescription(self):
        return self.description

    # Retrieves the seller of item.
    def getSeller(self):
        return self.seller

    # Retrieves the total price of the item.
    def getPrice(self):
        return self.itemPrice

    # Retrieves the total shipping costs of the item.
    def getShipping(self):
        return self.shippingCost

    # Retrieves the item's ID number.
    def getID(self):
        return self.itemID

    # Retrieves the total quantity amount of stock available of the item.
    def getQuantity(self):
        return self.stockQuantity

    # Returns true or false depending whether the item's is available in-stock within the shop. 
    def checkStock(self):
        return self.inStock